concent_api/middleman/tests/test_middleman.py

```diff
@@ -119,69 +135,140 @@ def test_that_crash_of_the_server_is_reported_to_sentry(self):
+            timeout = 0.2
+            short_delay = 0.1
+            schedule_sigterm(delay=timeout)
+            fake_client = socket.socket()
+            client_thread = get_client_thread(send_data, fake_client,  self.data_to_send, self.internal_port, short_delay)
+            client_thread.start()
+
+            error_message = "Connection_error"
+
+            with mock.patch.object(self.middleman, "_respond_to_user", side_effect=Exception(error_message)):
+                with pytest.raises(SystemExit):
+                    self.middleman.run()
+            client_thread.join(timeout)
+            fake_client.close()
+            self.crash_logger_mock.error.assert_called_once()
+            assert_that(self.crash_logger_mock.error.mock_calls[0][1][0]).contains(error_message)
 
     def test_that_broken_connection_from_signing_service_is_reported_to_sentry(self):
-        timeout = 0.2
-        short_delay = 0.1
-        schedule_sigterm(delay=timeout)
-        fake_client = socket.socket()
-        client_thread = get_client_thread(send_data, fake_client, "Oi!\n", self.external_port, short_delay)
-        client_thread.start()
-
-        error_message = "Connection_error"
-
-        with mock.patch.object(self.middleman, "_respond_to_user", side_effect=Exception(error_message)):
-            with pytest.raises(SystemExit):
-                self.middleman.run()
-        client_thread.join(timeout)
-        fake_client.close()
-        self.crash_logger_mock.error.assert_called_once()
-        assert_that(self.crash_logger_mock.error.mock_calls[0][1][0]).contains(error_message)
+        with override_settings(
+            CONCENT_PRIVATE_KEY=CONCENT_PRIVATE_KEY,
+            CONCENT_PUBLIC_KEY=CONCENT_PUBLIC_KEY,
+        ):
+            timeout = 0.2
+            short_delay = 0.1
+            schedule_sigterm(delay=timeout)
+            fake_client = socket.socket()
+            client_thread = get_client_thread(send_data, fake_client,  self.data_to_send, self.external_port, short_delay)
+            client_thread.start()
+
+            error_message = "Connection_error"
+
+            with mock.patch.object(self.middleman, "_respond_to_user", side_effect=Exception(error_message)):
+                with pytest.raises(SystemExit):
+                    self.middleman.run()
+            client_thread.join(timeout)
+            fake_client.close()
+            self.crash_logger_mock.error.assert_called_once()
+            assert_that(self.crash_logger_mock.error.mock_calls[0][1][0]).contains(error_message)
+
+    def test_that_broken_escaping_from_signing_service_is_handled(self):
+        with override_settings(
+            CONCENT_PRIVATE_KEY=CONCENT_PRIVATE_KEY,
+            CONCENT_PUBLIC_KEY=CONCENT_PUBLIC_KEY,
+        ):
+            timeout = 0.2
+            short_delay = 0.1
+            schedule_sigterm(delay=timeout)
+            connections = Connections()
+            exception = BrokenEscapingInFrameMiddlemanProtocolError
```

Comment:
this block (ending on `expected_data =`) should be extracted to a function like `get_raw_error_frame_bytes` or sth similar and reused in the tests

Reply:
Done